### ISSUE TITLE
needless_doctest_main reported even though code block is marked no_test

### ISSUE THREAD

**Issue by udoprog** [rust-lang/rust-clippy#10491]
### Summary

Not sure if intended or not, but a block marked as `no_test` should probably be considered a verbatim code block rather than Rust code to be analyzed:

### Lint Name

needless_doctest_main 

### Reproducer

```rust
//! ```rust,no_test
//! use std::collections::HashMap;
//!
//! fn main() {
//!     let mut m = HashMap::new();
//!     m.insert(1u32, 2u32);
//! }
//! ```
```

### Version

```text
rustc 1.68.0 (2c8cc3432 2023-03-06)
binary: rustc
commit-hash: 2c8cc343237b8f7d5a3c3703e3a87f2eb2c54a74
commit-date: 2023-03-06
host: x86_64-pc-windows-msvc
release: 1.68.0
LLVM version: 15.0.6
```
